validator/client/beacon-api/handlers_test.go

```diff
@@ -73,6 +93,32 @@ func invalidJsonErrHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func createStateValidatorsHandler(t *testing.T, data []*rpcmiddleware.ValidatorContainerJson, pubKeys ...[]string) http.HandlerFunc {
```

Comment:
`pubKeys` is not needed - you don't use it in any meaningful way.

Reply:
`pubkeys` is used for example here:

https://github.com/nalepae/prysm/blob/0f929241f7c2c555ed9a07a9999ce96e121466aa/validator/client/beacon-api/index_test.go#L39

If, at the same time, I comment this line `[]string{stringPubKey},` AND I change in Validator Client implementation
the `params` variable, like in the following example:

```go
func (c beaconApiValidatorClient) getStateValidator(pubkey string) (*rpcmiddleware.StateValidatorsResponseJson, error) {
	// params := neturl.Values{"id": []string{pubkey}}
       params := neturl.Values{"<A WRONG KEY>": []string{pubkey}}

	url := buildURL(
		c.url,
		"eth/v1/beacon/states/head/validators",
		params,
	)
```

then the test will still success.

If I only modify the valdiator implementation without commenting the `[]string{stringPubKey},` line in the test, then the test will fail.

This `pubKeys` variable allow us to ensure the Validator Client calls the Beacon Node with the well suited URL.

Reply:
Anyway, using GoMock instead.